src/main/kotlin/xyz/avalonxr/command/WelcomeCommand.kt

```diff
@@ -0,0 +1,160 @@
+            "preview" -> previewSubcommand(source, options)
+            else -> welcomeMessageError(CommandError.CommandNotFound("[set, preview]"), source)
+        }
+    }
+
+    private fun setSubcommand(
+        source: ChatInputInteractionEvent,
+        options: OptionStore
+    ): CommandResult {
+        val guildId = source.interaction.guildId
+            .map(Snowflake::asLong)
+            .getOrNull()
+            ?: return welcomeMessageError(CommandError.CommandNotFromGuild, source)
+        val defaultChannel = source.interaction.channel
+            .block()
+        val channel = when ("channel" in options) {
+            true ->
+                options
+                    .findChannel<TextChannel>("channel")
+                    ?: return welcomeMessageError(CommandError.IncorrectChannelType, source)
+            else -> defaultChannel
+        }
+        val message = options
+            .getOrDefault<String>("message", "Welcome!")
+        val welcomeMessage = messageRepository
+            .findByGuildId(guildId)
+            .getOrDefault(WelcomeMessage(guildId, channel.id.asLong(), message))
+
+        welcomeMessage.message = message
+        welcomeMessage.channelId = channel.id.asLong()
+        messageRepository.save(welcomeMessage)
+
+        val channelName = channel.mention
+
+        source.reply(
+            "Configured welcome message in $channelName"
+        )
+            .withEphemeral(true)
+            .block()
+
+        return CommandResult.success()
+    }
+
+    private fun previewSubcommand(
+        source: ChatInputInteractionEvent,
+        options: OptionStore
+    ): CommandResult {
+        val formatted = options
+            .getOrDefault<Boolean>("formatted", false)
+
+        val guildId = source.interaction.guildId
+            .map(Snowflake::asLong)
+            .getOrNull()
+            ?: return welcomeMessageError(CommandError.CommandNotFromGuild, source)
+
+        val welcomeMessage = messageRepository
+            .findByGuildId(guildId)
+            .getOrNull()
+        val content = welcomeMessage
+            ?.message
+            ?: "No welcome message configured."
+        val channelName = welcomeMessage
+            ?.channelId
+            ?.let { source.interaction.guild.block()?.getChannelById(Snowflake.of(it))?.block() }
+            ?.mention
+
+        val message = when (formatted) {
+            true -> content
+            else -> content
+        }
+
+        val template = "$message in channel: $channelName"
+
+        source.reply(template)
+            .withEphemeral(true)
+            .block()
+
+        return CommandResult.success()
+    }
+
+    private fun welcomeMessageError(
+        error: AvalonError,
+        event: ChatInputInteractionEvent
+    ): CommandResult {
+        event.reply(error.message)
+            .withEphemeral(true)
+            .block()
+
+        return CommandResult.failure(error)
+    }
```

Comment:
This can work for the moment, but should we maybe instead be sending a message via default command handler instead to avoid needing to do this explicitly each time?

Reply:
Solved by #45